src/physics/src/scalar_ode.C

```diff
@@ -56,35 +56,44 @@ namespace GRINS
     // we've clearly got a System to grab hold of with all it's
     // variables initialized.
 
-    libMesh::ParsedFEMFunction<libMesh::Number> *tdf
-      (new libMesh::ParsedFEMFunction<libMesh::Number> (*system, ""));
-    this->time_deriv_function.reset(tdf);
+    {
+      std::unique_ptr<libMesh::ParsedFEMFunction<libMesh::Number>> tdf =
+        libmesh_make_unique<libMesh::ParsedFEMFunction<libMesh::Number>>(*system, "");
```

Comment:
I'm still not a fan of using `auto` *everywhere*, but it's nice for idioms like this.

Reply:
Oh that's a good point. I also tend to err on the side of not using `auto` but agreed this could be a good place for it. It didn't trip my annoyance like iterators do (which is where I use `auto` typically).

Reply:
Add `auto` here in d173d7ac